src/System.ServiceModel.Federation/src/System/ServiceModel/Federation/WSTrustChannelFactory.cs

```diff
@@ -0,0 +1,90 @@
+        public WSTrustChannelFactory(ServiceEndpoint endpoint)
+            : base(endpoint)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WSTrustChannelFactory" /> class associated with a specified
+        /// name for the endpoint configuration and remote address.
+        /// </summary>
+        /// <param name="endpointConfigurationName">The configuration name used for the endpoint.</param>
+        /// <param name="remoteAddress">The <see cref="EndpointAddress" /> that provides the location of the service.</param>
+        public WSTrustChannelFactory(string endpointConfigurationName, EndpointAddress remoteAddress)
+            : base(endpointConfigurationName, remoteAddress)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WSTrustChannelFactory" /> class with a specified binding
+        /// and endpoint address.
+        /// </summary>
+        /// <param name="binding">The <see cref="Binding" /> specified for the channels produced by the factory</param>
+        /// <param name="remoteAddress">The <see cref="EndpointAddress" /> that provides the location of the service.</param>        
+        public WSTrustChannelFactory(Binding binding, EndpointAddress remoteAddress)
+            : base(binding, remoteAddress)
+        {
+        }
+
+        /// <summary>
+        /// Gets or sets the version of WS-Trust the created channels will use for serializing messages.
+        /// </summary>
+        /// <remarks>
+        /// Default is <see cref="WsTrustVersion.Trust13"/>
+        /// </remarks>
+        public WsTrustVersion TrustVersion { get; set; } = WsTrustVersion.Trust13;
+
+        /// <summary>
+        /// Creates a <see cref="WSTrustChannel" /> that is used to send messages to a service at a specific 
+        /// endpoint address through a specified transport address.
+        /// </summary>
+        /// <param name="address">The <see cref="EndpointAddress" /> that provides the location of the service.</param>
+        /// <param name="via">The <see cref="Uri" /> that contains the transport address to which the channel sends messages.</param>
+        /// <returns></returns>
+        public override IWSTrustChannelContract CreateChannel(EndpointAddress address, Uri via)
+        {
+            return CreateTrustChannel(base.CreateChannel(address, via) as IRequestChannel);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="WSTrustChannel" /> using parameters that reflect the configuration of
+        /// this factory.
+        /// </summary>
+        /// <param name="requestChannel">The <see cref="IRequestChannel"/> that will use used to send and receive messages.</param>
+        /// <returns></returns>
+        protected virtual WSTrustChannel CreateTrustChannel(IRequestChannel requestChannel)
+        {
+            return new WSTrustChannel(this, requestChannel, TrustVersion);
+        }
```

Comment:
As WSTrustChannel has such an odd constructor where it is passed this class basically just to get the MessageVersion from the binding and the EndpointAddress uri, what are your thoughts of making the constructors on WSTrustChannel internal? They aren't really there for public consumption really, the expected use case is to get an instance of WSTrustChannel from WSTrustChannelFactory.

Reply:
We could certainly do that that would limit the ability for any custom overrides of  IssueAsync.
We could wait until we are asked for the overrides, you are closer to consumers at this point in time, what do you think?

Reply:
Then make the constructor internal protected. That still stops direct instantiation but allows derivation.

Reply:
Or another option is to not do exactly the same as NetFx but instead pass the things which the channel factory is actually used for. We could change the constructor to take a MessageVersion and a Uri. This way you don't need multiple channel factory instances if your use case is to talk to multiple STS's with everything else the same. ChannelFactory.CreateChannel has an overload to create a channel for a specific endpoint so I'm going to see if there's a reliable way to extract the endpoint address from the requestChannel as that would be preferable.

Reply:
@mconnew i am all for that, i originally wanted to keep the API's as close as possible..
But, given that users will need to make some code changes, i agree there is a simpler and more natural model.
For example, the WSTrustChannelFactory shouldn't need to know what trust version is used.
The model in NetFx required a version specific WsTrustSerializer, however the current work doesn't require that.
I agree if we had the endpoint address, the api would be simpler.

Reply:
Simplified and removed methods